media/js/base/mozilla-traffic-cop.js

```diff
@@ -0,0 +1,174 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// create namespace
+if (typeof Mozilla === 'undefined') {
+    var Mozilla = {};
+}
+
+/**
+ * Traffic Cop traffic redirector for A/B/x testing
+ *
+ * Example usage:
+ *
+ * var cop = new Mozilla.TrafficCop({
+ *     id: 'exp_firefox_new_all_link',
+ *     variations: {
+ *         'v=1': 25,
+ *         'v=2': 25,
+ *         'v=3': 25
+ *     }
+ * });
+ *
+ * cop.init();
+ *
+ *
+ * @param Object config: Object literal containing the following:
+ *      String id (required): Unique-ish string for cookie identification.
+ *          Only needs to be unique to other currently running tests.
+ *      Object variations (required): Object holding key/value pairs of
+ *          variations and their respective traffic percentages. Example:
+ *
+ *          variations: {
+ *              'v=1': 20,
+ *              'v=2': 20,
+ *              'v=3': 20
+ *          }
+ */
+Mozilla.TrafficCop = function(config) {
+    'use strict';
+
+    // make sure config is an object
+    config = (typeof config === 'object') ? config : {};
+
+    // store id
+    this.id = config.id;
+
+    // store variations
+    this.variations = config.variations;
+
+    // store total percentage of users targeted
+    this.totalPercentage = 0;
+
+    this.redirectVariation;
+
+    // calculate and store total percentage of variations
+    for (var v in this.variations) {
+        this.totalPercentage += this.variations[v];
```

Comment:
As `config` is passed in, should we wrap this in a conditional that checks `hasOwnProperty` to prevent looping over all enumerable properties (just for good measure)? e.g.

```
if (this.variations.hasOwnProperty(v)) {
    this.totalPercentage += this.variations[v];
}
```